File: Byte&Buy.AngularSPA/src/app/features/profile/api-dto/shipping-address-add-request.ts
import { Guid } from "guid-typescript";

export interface ShippingAddressAddRequest {
    countryId: Guid;
    label: string;
    street: string;
    houseNumber: string;
    postalCity: string;
    postalCode: string;
    city: string;
    country: string;
    flatNumber: string | null;
    isDefault: boolean;
}

File: Byte&Buy.Backend/src/ByteBuy.Core/DTO/Public/ShippingAddress/ShippingAddressAddRequest.cs
using System.ComponentModel.DataAnnotations;

namespace ByteBuy.Core.DTO.Public.Address;

public record ShippingAddressAddRequest(
    [Required] Guid CountryId,
    [Required, MaxLength(50)] string Label,
    [Required, MaxLength(50)] string Street,
    [Required, MaxLength(10)] string HouseNumber,
    [Required, MaxLength(20)] string PostalCode,
    [Required, MaxLength(50)] string PostalCity,
    [Required, MaxLength(50)] string City,
    [MaxLength(10)] string? FlatNumber,
    [Required] bool IsDefault
    );
